src/charm.py

```diff
@@ -17,6 +17,13 @@
 )
 
 
+class CustomEvent(HookEvent):
```

Comment:
Apart from the general discussion of how to handle setting the status, we might want to look into using a decorator if we want to run a function after a hook. Something like:

```python
@postscript(self._handle_custom_event)
def main(self, event):
    ...
```

The implementation of the decorator would be a little magic, but there's also lots of learning resources for bits of Python magic like that.